internal/service/lightsail/database.go

```diff
@@ -0,0 +1,462 @@
+package lightsail
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+func ResourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: ResourceDatabaseCreate,
+		Read:   ResourceDatabaseRead,
+		Update: ResourceDatabaseUpdate,
+		Delete: ResourceDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			State: ResourceDatabaseImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"master_database_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"master_password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"master_username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"blueprint_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"bundle_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			// Optional attributes
+			"preferred_backup_window": {
```

Comment:
Input validation should be added to match the AWS Public API.
In this case:

1. Must be in the ddd:hh24:mi-ddd:hh24:mi format.
2. Valid days: Mon, Tue, Wed, Thu, Fri, Sat, Sun.
3. Must be at least 30 minutes.
4. Specified in Coordinated Universal Time (UTC).
5. Example: Tue:17:00-Tue:17:30

You can see a good example of validation function that evaluates multiple conditions here:
[terraform-provider-aws/internal/service/lexmodels/bot.go](https://github.com/hashicorp/terraform-provider-aws/blob/c942e1dbacc12e4f53ae4ab61bdf1fa0e417f5a2/internal/service/lexmodels/bot.go#L115)

 
AWS Lightsail API docs
https://docs.aws.amazon.com/lightsail/2016-11-28/api-reference/API_CreateRelationalDatabase.html#Lightsail-CreateRelationalDatabase-request-preferredBackupWindow

Reply:
For this I used the existing validation function that is currently being used by the `rds_cluster` resource.  I also found an existing validation function to validate the `preferred_maintenance_window`.